ansible/roles/fyrevm_provision/tasks/fyrevm_provision.yml

```diff
@@ -55,57 +89,117 @@
+      uri:
+        url: "https://api.fyre.ibm.com/rest/v1/?operation=query&request=showrequests&request_id={{ request_id | string }}"
+        method: GET
+        user:  "{{ fyreuser }}"
+        validate_certs: false
+        password: "{{ fyreapikey }}"
+        force_basic_auth: true
+        return_content: true
+      changed_when: false
+      register: request_status_response
+      delay: 20
+      retries: 600
+      until: (request_status_response.status == 200) and request_status_response.content is defined and not (request_status_response.content | from_json)['request'][0].status is match('.*building.*')
+
+    - name: Check for errors
+      fail: msg="cluster build request error"
+      when: (request_status_response.content | from_json)['request'] | map(attribute='status') | unique | difference(['completed','deployed']) | length > 0
+
+    - name: Fetch the cluster status
+      uri:
+        url: "https://api.fyre.ibm.com/rest/v1/?operation=query&request=showclusters"
+        method: GET
+        user:  "{{ fyreuser }}"
+        validate_certs: false
+        password: "{{ fyreapikey }}"
+        force_basic_auth: true
+        return_content: true
+      changed_when: false
+      register: cluster_status_response
+      until: cluster_status_response.status != -1
+
+    - name: Set cluster_status_json fact
+      set_fact:
+        cluster_status_json: "{{ item }}"
+      loop: "{{ (cluster_status_response.content | from_yaml)['clusters'] }}"
+      when: item.name == clusterName
+
+    - name: Set cluster_status fact
+      set_fact:
+        cluster_status: "{{ cluster_status_json['status'] | default('error') }}"
+
+    - name: Check for error
+      assert:
+        that: cluster_status != 'error'
+
+- name: Fetch the cluster details
   uri:
-    url: "{{fyre_clusterstatusurl}}"
+    url: "https://api.fyre.ibm.com/rest/v1/?operation=query&request=showclusterdetails&cluster_name={{ clusterName }}"
     method: GET
     user:  "{{ fyreuser }}"
+    validate_certs: false
     password: "{{ fyreapikey }}"
-    force_basic_auth: True
-    validate_certs: False
-    return_content: yes
-    body_format: json
+    force_basic_auth: true
+    return_content: true
   changed_when: false
-  register: clusterStatus
-  until: (clusterStatus.status == 200) and clusterStatus.json is defined and (clusterStatus.json[clusterName][0].status  == "deployed" or clusterStatus.json[clusterName][0].status == "error")
-  retries: 600
-  delay: 20
-
-- name: check for error status after loop check
-  fail:
-   msg:
-   - "{{clusterStatus.json[clusterName][0]}}"
-  when:
-  - clusterStatus.json is defined
-  - (clusterStatus.json[clusterName][0].status) == 'error'  
+  register: cluster_details_response
+  until: cluster_details_response.status != -1
+
+- name: Set fact for cluster_details_json
+  set_fact:
+    cluster_details_json: "{{ (cluster_details_response.content | from_json)[clusterName] }}"
```

Comment:
Would like to avoid leaking cluster_details_json beyond this role, a block scoped var might be more appropriate.

Reply:
that was from my stuff.  rather than choosing which subset of those details other roles might want I created that var.  the difference is that in my stuff that is done under a 'cluster-facts' hostvar and is not leaked into any other context.  given that all vars are running in the same context in community-automation I agree that it is not appropriate here and should use block scope as you suggest